bot/university/university.py

```diff
@@ -0,0 +1,378 @@
+"""Contains a Cog for all functionality regarding our University."""
+
+import xml.etree.ElementTree as ET
+import re
+from typing import Dict, List, Optional, Union
+from datetime import datetime
+import requests
+import discord
+from discord.ext import commands
+from bot.util import SelectionEmoji
+from bot.persistence import DatabaseConnector
+from bot import constants
+
+
+class UniversityCog(commands.Cog):
+    """Cog for Functions regarding our university."""
+
+    def __init__(self, bot):
+        """Initializes the Cog.
+
+        Args:
+            bot (discord.ext.commands.Bot): The bot for which this cog should be enabled.
+        """
+        self.bot = bot
+        self._db_connector = DatabaseConnector(constants.DB_FILE_PATH, constants.DB_INIT_SCRIPT)
+
+    @commands.group(name="ufind")
+    async def ufind(self, ctx: commands.Context):
+        """Command Handler for the `ufind` command.
+
+        Allows users to search the service `ufind` for various information regarding staff members, courses and exams
+        and posts them on Discord. For each of these categories exists a corresponding subcommand. If no subcommand has
+        been invoked, a help message for this command will be posted instead.
+
+        Args:
+            ctx (discord.ext.commands.Context): The context in which the command was called.
+        """
+        if ctx.invoked_subcommand is None:
+            await ctx.send_help(ctx.command)
+
+    @ufind.command(name='staff')
+    async def ufind_get_staff_data(self, ctx: commands.Context, *, search_term: str):
```

Comment:
Be aware that if you add the @command_log annotation the log message will label the command as 'ufind_get_staff_data' instead of 'ufind staff'. Just for your interest.

Reply:
Will be changed